packages/template-retail-react-app/app/utils/product-utils.js

```diff
@@ -35,19 +35,39 @@ export const getDisplayVariationValues = (variationAttributes, values = {}) => {
 }
 
 /**
- * This function extract the promotional price from a product. If there are more than one price, the smallest price will be picked
+ * This function extract the list price and current price of a product
+ * If a product has promotional price, it will prioritize that value for current price
+ * List price will take the highest value among the price book prices
  * @param {object} product - product detail object
- * @returns {{discountPrice: number, basePrice: number | string}}
+ * @param {object} opts - product detail object
+ * @returns {{listPrice: number, currentPrice: number}}
  */
-export const getDisplayPrice = (product) => {
-    const basePrice = product?.pricePerUnit || product?.price
+export const getDisplayPrice = (product, opts = {}) => {
+    const {quantity = 1} = opts
     const promotionalPriceList = product?.productPromotions
         ?.map((promo) => promo.promotionalPrice)
         .filter((i) => i !== null && i !== undefined)
-    // choose the smallest price among the promotionalPrice
-    const discountPrice = promotionalPriceList?.length ? Math.min(...promotionalPriceList) : null
+    const promotionalPrice = promotionalPriceList?.length ? Math.min(...promotionalPriceList) : null
+
+    // if a product has tieredPrices, get the tiered that has the closest quantity to current quantity
+    const closestTieredPrice = product?.tieredPrices?.reduce((prev, curr) => {
+        return Math.abs(curr.quantity - quantity) < Math.abs(prev.quantity - quantity) ? curr : prev
+    })
+    const salePrices = [closestTieredPrice?.price, promotionalPrice, product?.price].filter(Boolean)
+    // pick the smallest price among these price for the "current" price
+    let currentPrice = salePrices?.length ? Math.min(...salePrices) : 0
+
+    let tieredPrices = product?.priceRanges || product?.tieredPrices || []
```

Comment:
I don't think it's a good idea to mix the `priceRanges` and the `tieredPrices` objects, looking at the API they have similar properties but they aren't exactly the same. This makes it a little hard to read what is happening in in function because it's possible you are calling a price range object a tiered price object.

Reply:
Good point, I've separated the logic and gave it some context

Reply:
Agree. They both are different objects. AFAIK tiered prices are mainly for different quantities. If we have decided to display the range may be we can just use priceRanges?

Reply:
priceRanges only available for master product or a product set. For variant product/standard/bundle, we only have tieredPrices. yes, tiredPrices are mainly for different quantities, but i assume they always have the first tiered which is for quantity 1 before they have another tiered. The highest value is the list seems to be the list price but I could be wrong. 🤔